src/lib/HST/Effect/Report.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE TemplateHaskell, ScopedTypeVariables #-}
+{-# LANGUAGE LambdaCase, BlockArguments #-}
+
+-- | This module defines an effect for computations that can report 'Message's
+--   such as errors and warnings at runtime.
+--
+--   There are handlers for such computations that either collect the reported
+--   messages or print them to a file 'Handle'. Reported messages can be either
+--   fatal or non-fatal. The usual interpretation of a fatal message is for
+--   the computation to be 'cancel'ed.
+
+module HST.Effect.Report
+  ( -- * Messages
+    Severity(..)
+  , Message(..)
+    -- * Effect
+  , Report
+    -- * Actions
+  , report
+  , reportFatal
+    -- * Interpretations
+  , runReport
+  , reportToOutputOrCancel
+  , reportToHandleOrCancel
+  , filterReportedMessages
+    -- * Interpretations for Other Effects
```

Comment:
This is the same capitalization as used by Polysemy (see [`Polysemy.Trace`](https://hackage.haskell.org/package/polysemy-1.3.0.0/docs/Polysemy-Trace.html#g:4) for example). This is also consistent with the [capitalization rules for Markdown headings](https://github.com/FreeProving/guidelines/blob/main/CONTRIBUTING.md#use-atx-headings) in our styleguide.

> - Capitalize the first word of the heading (e.g. "The CI Pipeline" instead of "the CI Pipeline").
> - Capitalize all other words except for articles, conjunctions and prepositions (e.g. "How Can I Contribute?" instead of "How can I contribute?" but "Haskell to Coq" instead of "Haskell To Coq").

"Other" has to be capitalized since is neither an article, conjunction nor preposition. "For" on the other hand is not capitalized since it is a preposition.

Reply:
Interesting, I didn't know about this capitalization rule yet.